src/components/Card/Card.tsx

```diff
@@ -1,18 +1,24 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import { Typography } from '@mui/material';
 import classNames from 'classnames';
 import PlayCircleIcon from '@mui/icons-material/PlayCircle';
 import { TypeMovieCard } from '../types';
 import style from './Card.module.scss';
-import { useAppSelector } from '../../store/store';
+import { useAppDispatch, useAppSelector } from '../../store/store';
+import { fetchGenresData } from '../../store/reducers/CardsActions';
 
 function Card(movieData: TypeMovieCard) {
   const { title, vote_average, poster_path, genre_ids, actorClass, id } = movieData;
   const { genresArray } = useAppSelector((state) => state.genres);
```

Comment:
Let's pass genresArray as a prop and use the Card component as just UI representation without logic